main/jlp/widgets/widget_factory.cpp

```
@@ -2480,6 +2497,294 @@ lv_obj_t* build_voice(BuildCtx& ctx, JsonObjectConst spec, std::string* err) {
+        c->sh->alive.store(false);
+        if (c->started) stream_client_stop();
+        if (c->udp_sock >= 0) close(c->udp_sock);
+        delete c;
+      },
+      LV_EVENT_DELETE, nullptr);
+
+  return root;
+}
+
+void stream_widgets_notify_visibility(lv_obj_t* container, bool visible) {
+  if (!container) return;
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)